lib/fx/adapters/postgres/functions.rb

```diff
@@ -39,7 +42,12 @@ def initialize(connection)
         #
         # @return [Array<Fx::Function>]
         def all
-          functions_from_postgres.map { |function| to_fx_function(function) }
+          functions_from_postgres.map do |function|
+            to_fx_function(
+              "name" => schema_aware_name(function),
+              "definition" => schema_aware_definition(function)
```

Comment:
Thanks for the nudge @teoljungberg! I'm also a `scenic` user and so I dug into how that gem works, and ended up proposing this change on their end (which so far seems to be well-received, though I cannot say whether it will ultimately merge): https://github.com/scenic-views/scenic/pull/401

In resolving some discrepancies within that gem, I ended up relying on the way that a core Rails contribution enabled PostgreSQL-handling for enums: https://github.com/rails/rails/pull/45740

I think it would make a ton of sense for _both_ `fx` and `scenic` to behave similar to this core `enum` approach, which is roughly as follows:
1. Query for all user-created structures that are accessible to the current PG user (regardless of search_path)
2. If the structure's schema does not match `current_schema()`, include the fully-qualified name in `schema.rb`
3. If the structure's schema _does_ match `current_schema()`, do not include the schema in the generated `schema.rb` name.

Points 2 and 3 there are really crucial. This allows a `schema.rb` to be shared across contexts where the "default" search path might change (and is entirely dependent on external/config factors -- it is not guaranteed to be `public`, and some PAAS providers do not even place this in the end-developer's control), while still allowing developers to explicitly specify non-default schemas and have those survive the round-trip to `schema.rb`.